samples/Matty/Matty.Client/Program.cs

```diff
@@ -0,0 +1,121 @@
+            Console.ReadLine();
+
+            using var client = new HttpClient();
+
+            try
+            {
+                // Make call to "/.well-known/openid-configuration" endpoint
+                // to discover identity server configuration document
+                var discovery = await client.GetDiscoveryDocumentAsync("https://localhost:44321");
+                if (discovery.IsError)
+                {
+                    throw new Exception(discovery.Error);
+                }
+
+                var deviceResponse = await client.RequestDeviceAuthorizationAsync(new DeviceAuthorizationRequest
+                { 
+                    Address = discovery.DeviceAuthorizationEndpoint,
+                    Scope = "openid offline_access profile email",
+                    ClientId = "device",
+                });
+                if (deviceResponse.IsError)
+                {
+                    throw new Exception(deviceResponse.Error);
+                }
+
+                Console.WriteLine("+-------------------------------------------------------------------+");
+                Console.WriteLine("Device endpoint response");
+                Console.WriteLine($" - Device code: {deviceResponse.DeviceCode}");
+                Console.WriteLine($" - User code: {deviceResponse.UserCode}");
+                Console.WriteLine($" - Verification uri: {deviceResponse.VerificationUri}");
+                Console.WriteLine($" - Verification uri (complete): {deviceResponse.VerificationUriComplete}");
+
+                Console.WriteLine("+-------------------------------------------------------------------+");
+                Console.WriteLine(" - Please open the verification uri in the browser");
+                Console.WriteLine(" and enter the user code to authorize this device ");
+                Console.WriteLine("+-------------------------------------------------------------------+");
+                Console.Write(" - Please press Enter to proceed...");
+                Console.ReadLine();
+                TokenResponse tokenResponse;
+                do
+                {
+                    tokenResponse = await client.RequestDeviceTokenAsync(new DeviceTokenRequest
+                    {
+                        Address = discovery.TokenEndpoint,
+                        ClientId = "device",
+                        DeviceCode = deviceResponse.DeviceCode
+                    });
+
+                    if (tokenResponse is { IsError: true, Error: Errors.AuthorizationPending })
+                    {
+                        Console.WriteLine(" - authorization pending...");
+
+                        // Note: `deviceResponse.Interval` is the minimum number of seconds
+                        // the client should wait between polling requests.
+                        // In this sample the client will retry every 60 seconds at most.
+                        await Task.Delay(Math.Min(deviceResponse.Interval, 60) * 1000);
```

Comment:
Actually, I'm wondering if we shouldn't use `Math.Clamp(deviceResponse.Interval, 1, 60)` instead, to guard against negative values, just in case:

```suggestion
                        // In this sample the client will retry every 60 seconds at most.
                        await Task.Delay(Math.Clamp(deviceResponse.Interval, 1, 60) * 1000);
```

Reply:
I think using Clamp as a guard is a good idea, there is no guard in the `DeviceAuthorizationResponse`, just setting 5 as default.
https://github.com/IdentityModel/IdentityModel/blob/44cc1f98afa9cc4efc9acd8be866e03c9692a2c8/src/Client/Messages/DeviceAuthorizationResponse.cs#L58

The identity server should protect this value setter in the response from being negative in the first place.
I tried to find how this value is configured in `openIddict`, I could not find it.